app/controllers/group_assignment_invitations_controller.rb

```diff
@@ -52,6 +52,10 @@ def authorize_group_access
     group_id = group_params[:id]
 
     return unless group_id.present?
+    group = Group.find(group_id)
```

Comment:
@beydogan I this logic is getting to be a bit much, I think we need to start extracting this out into different methods.

What do you think about something like this?

```ruby
def authorize_group_access
  group_id = group_params[:id]

  return unless group_id.present?
  return unless group_apart_of_grouping?(group_assignment, group_id)
  return unless group_has_reach_max_members?(group_assignment, group_id)
end
```